src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -26,7 +26,7 @@ private static IEnumerable<DkmClrModuleInstance> GetModulesInAppDomain(this DkmC
             var appDomainId = appDomain.Id;
             return runtime.GetModuleInstances().
                 Cast<DkmClrModuleInstance>().
-                Where(module => module.AppDomain.Id == appDomainId);
+                Where(module => !module.AppDomain.IsUnloaded && module.AppDomain.Id == appDomainId);
```

Comment:
I agree that there's probably a race (I guess we need a try-catch?) but I'm not sure how we could perform the check up-front since it seems to depend on the iteration variable.